added condor queue selection and normalErrs, fixed hadd multithreading

mlizzo opened the pull request:
Ciao @giorgiopizz , @dittmer , @NTrevisani 

I've added a small set of features, like the possibilty to select the job flavour queue and change it back when resubmitting failed jobs (-q option, default is "workday"). Standard printout obtained when a macro is compiled is not interpreted as an error anymore, and the hadd step now runs over 10 cores (hard-coded, but it can be easily put as an option if needed). Moreover, I've commented out the implicitMT RDF mode, which I don't think is needed when jobs are submitted to condor and sometimes it causes troubles. Finally, an auxiliary variable is set when the code is installed, which points to the "start.sh" script, without assuming any relative path. This is important, because, as of now, configurations in batch mode only work if they're two levels below where the start.sh is located, which is not ideal (see https://github.com/latinos/mkShapesRDF/blob/master/mkShapesRDF/shapeAnalysis/BatchSubmission.py#L114).

Please have a look at it, and let me know if you have some feedback/way to improve these features - assuming they're not bugs 
Cheers,
Mattia

lviliani commented:
Just for reference, the issue related to the `start.sh` script is also reported here https://github.com/latinos/mkShapesRDF/issues/8

giorgiopizz commented on mkShapesRDF/shapeAnalysis/mkShapesRDF.py:
```diff
@@ -353,7 +367,7 @@ def normalErrsF(k):
         print(f"Hadding files into {folder}/{outputFolder}/{outputFile}")
         process = subprocess.Popen(
-            f'hadd -j {folder}/{outputFolder}/{outputFile} {" ".join(filesToMerge)}',
+            f'hadd -j 10 {folder}/{outputFolder}/{outputFile} {folder}/{outputFolder}/{outputFileTrunc}__ALL__*.root',
```
if we change the samples to use in a configuration (e.g. limit to a couple of samples for debugging) the output root files might be stored in the same outputFolder. With this syntax you would `hadd` also old files. Was this related to the issue of `hadding` a very long list of root files? We might thing of writing `filesToMerge` to a txt file and use that for the had

mlizzo replied:
mmm I see your point but it seems like a corner case to me. One could simply adjust the "tag" in the configuration.py file for debugging purposes, in order to keep configurations uncorrelated with each other. Does this make sense to you? Otherwise we could think of another solution

mlizzo commented on PlotsConfigurationsRun3:
This is something I forgot to mention. I think having [PlotsConfigurationsRun3](https://github.com/latinos/PlotsConfigurationsRun3) as a submodule of mkShapesRDF would be nice, so that one can download RDF/Run3 configs together with the code (if explicitly enabled). I don't have a strong opinon on this though, so feel free to discard the change

mlizzo commented:
Ciao @giorgiopizz , I've removed the submodule dependency to the PlotsConfigurationsRun3 repository and fixed the arg_max bash limit as you suggested, it should work properly but please have a look at it. Let me know if you think there's something else I should modify

mlizzo commented:
Ping

mlizzo commented:
Sorry, I just added a last-minute commit as I realized the "queue" argument was missing from the submit call, therefore it kept using the default value. Now it works fine, apparently I only updated the resubmit call in the beginning. Won't add any new modifications, apology for that :)
